libs/mrwebrtc/test/video_track_source_tests.cpp

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "interop_api.h"
+#include "video_frame.h"
+#include "video_track_source_interop.h"
+
+#include "test_utils.h"
+
+namespace {
+
+class VideoTrackSourceTests
+    : public TestUtils::TestBase,
+      public testing::WithParamInterface<mrsSdpSemantic> {};
+
+}  // namespace
+
+#if !defined(MRSW_EXCLUDE_DEVICE_TESTS)
+
+INSTANTIATE_TEST_CASE_P(,
+                        VideoTrackSourceTests,
+                        testing::ValuesIn(TestUtils::TestSemantics),
+                        TestUtils::SdpSemanticToString);
+
+TEST_P(VideoTrackSourceTests, CreateFromDevice) {
+  mrsLocalVideoDeviceInitConfig config{};
+  mrsVideoTrackSourceHandle source_handle{};
+  ASSERT_EQ(mrsResult::kSuccess,
+            mrsVideoTrackSourceCreateFromDevice(&config, &source_handle));
+  ASSERT_NE(nullptr, source_handle);
+  mrsVideoTrackSourceRemoveRef(source_handle);
+}
+
+// TODO - Don't use device, run outside MRSW_EXCLUDE_DEVICE_TESTS
+TEST_P(VideoTrackSourceTests, Name) {
```

Comment:
nit: 80 lines of test for every function that returns a string looks a bit silly, especially considering that the tested code is essentially the same. It would be nicer to have a `TestFunctionReturningString` helper.